content/en/docs/tasks/administer-cluster/cpu-management-policies.md

```diff
@@ -68,6 +68,23 @@ and `CPUManagerPolicyBetaOptions` feature gates. Diverging from the Kubernetes s
 feature gates guard groups of options, because it would have been too cumbersome to add a feature
 gate for each individual option.
 
+### Changing the CPU Manager Policy
+
+In order to change the CPU Manager policy on a node, perform the following steps:
+
+1. [Drain](/docs/tasks/administer-cluster/safely-drain-node) the node.
+2. Stop kubelet.
+3. Remove the old CPU manager state file `/var/lib/kubelet/cpu_manager_state`.
+4. Edit the kubelet configuration to change the CPU manager policy to the desired value.
+4. Reboot the node.
```

Comment:
It has to be a reboot? How come?

Reply:
I'm not sure of the specifics, but the KubeletConfiguration docs for changing `cpuManagerPolicy` tell you to do that: https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration

That advice came from this PR: https://github.com/kubernetes/kubernetes/pull/64354

But those docs are slated to be removed in 1.24: https://github.com/kubernetes/kubernetes/pull/106932/files#diff-e7cf3d9bacdf8ddf0db56a9a399993d2fc62897122206ecef17aee62c2e69d8fR342

Neither PR explicitly said why a reboot was necessary or if it still is. Just from looking at the PRs, I'm inclined to believe that a reboot _is_ still necessary (since mtaufen's PR was heavily reviewed, while updates to types.go in deprecation PR are likely to be skimmed-over).

If anyone in sig-node can clarify if a reboot is necessary for changing cpuManagerPolicy, that would be appreciated!

Reply:
I'm guessing that requesting a reboot is simpler to explain, but not actually needed. Ideally we'd document what you must actually restart.

For example, if you can:
```shell
systemctl stop kubelet.service
custom-cgroup-frob.sh # needs implementing - my point is that might be possible to do so
systemctl start kubelet.service
```
or
```shell
systemctl isolate emergency.target
# exit emergency environment
systemctl restart kubelet.target
```

then let's aim to document those steps (maybe by tracking a separate backlog issue).
Folks operating on bare metal may well value the extra detail.

Reply:
Reboot is not required — you can just remove the file `/var/lib/kubelet/cpu_manager_state` and restart your kubelet. This clears the state maintained by the CPUManager so that the cpu-sets set up by the new policy won’t conflict with it.

That said, if you do this, you may still have pods running that have cpu-sets set up for them by the old policy (and these may overlap with the cpu-sets that your new policy sets up).

If you are OK with this (or know this isn’t possible in your setup), then simply removing this file is OK.